v3/src/models/history/apply-model-change.ts

```diff
@@ -1,69 +1,29 @@
-import iframePhone from "iframe-phone"
 import { IAnyStateTreeNode } from "mobx-state-tree"
-import { DIMessage } from "../../data-interactive/iframe-phone-types"
+import { ICoreNotification } from "../../data-interactive/notification-core-types"
 import { ILogMessage } from "../../lib/log-message"
-import { getTileEnvironment } from "../tiles/tile-environment"
-import { withUndoRedoStrings } from "./codap-undo-types"
-import { withoutUndo } from "./without-undo"
+import { getHistoryService } from "./history-service"
 
-export interface INotification {
-  message: DIMessage
-  callback?: iframePhone.ListenerCallback
-}
 export type INotify =
-  INotification | (INotification | (() => Maybe<INotification>))[] | (() => Maybe<INotification | INotification[]>)
+  ICoreNotification |
+  (ICoreNotification | (() => Maybe<ICoreNotification>))[] |
+  (() => Maybe<ICoreNotification | ICoreNotification[]>)
```

Comment:
This type is redundantly defined in `history-service.ts`, so could be imported instead.